### DISCUSSION TITLE
ardusub_driver & mobile_to_maritime

### DISCUSSION THREAD

**Discussion by MichaelLeibbrand** [Robotic-Decision-Making-Lab/blue#178] [Q&A]
Hello, 
I followed all the installation instructions under https://robotic-decision-making-lab.github.io/blue/ 
and pulled the humble-desktop-nvidia image successfully. But if i want to start the simulation, integrate custom controllers or keyboard teleoperations I get following Errors:

~/ws_blue$ ros2 launch blue_demos bluerov2_demo.launch.yaml use_sim:=true
[INFO] [launch]: All log files can be found below /home/blue/.ros/log/2024-06-12-08-45-51-201624-michael-leibbrand-7269
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): "package 'ardusub_bringup' not found

~/ws_blue$ ros2 launch blue_demos bluerov2_controllers.launch.py use_sim:=true
[INFO] [launch]: All log files can be found below /home/blue/.ros/log/2024-06-12-08-43-37-416336-michael-leibbrand-7241
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): "package 'mobile_to_maritime' not found

I thought all these dependencies will be met by default if I have installed Blue with Docker ?

When I look it ardusub_driver https://github.com/Robotic-Decision-Making-Lab/ardusub_driver
it only supports ROS2 Iron? 

I need a working simulation environment  and want to try to avoid spending days setting it up and focus to implement my image based visual servoing control strategies.

Is the humble docker image not supporting those packages ?
Advice for best workflow for my project ?

ps. I can start the simulation with: 
ros2 launch blue_bringup bluerov2_heavy.launch.py use_sim:=true

**Comment by evan-palmer** [comment]
Thanks for checking out the project @MichaelLeibbrand! The issue that you are observing right now is #167. The PR that we are waiting on from the `control_msgs` devs was just merged a few days ago, and we are just waiting on the binaries to be published to close this issue

**Comment by evan-palmer** [comment]
Hey @MichaelLeibbrand, I wanted to let you know that the Humble branch should now be working. Let me know if you run into any issues with it!
